### PR TITLE
Favorite directories in file selector

### PR DESCRIPTION
**Pull request by marioba**
Add a section in file selection first activity with the favorite directories. 
To add a directory to the favorites, do a long click on the directory. 
To remove a directory from the favorites do a long click on the entry in the favorites section.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-507-dd2ca8d2a97934679b1732331aa41a177f1209d6-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-507-dd2ca8d2a97934679b1732331aa41a177f1209d6-x86.apk)

**Comment by qfield-fairy** [comment]
Uploaded test apks for [armv7](https://download.opengis.ch/qfield/ci-builds/qfield-dev-507-2a5ac724a6aca79b4a2b83a384835da29400191c-armv7.apk) and [x86](https://download.opengis.ch/qfield/ci-builds/qfield-dev-507-2a5ac724a6aca79b4a2b83a384835da29400191c-x86.apk)

File: android/res/layout/list_projects.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
               android:layout_width="match_parent" android:layout_height="match_parent">
 
-  <ListView android:id="@android:id/list" android:layout_width="match_parent"
+  <ListView android:id="@+id/list" android:layout_width="match_parent"
```
Comment (mbernasocchi) [inline android/res/layout/list_projects.xml]: just out of curiosity, what is the +?

Comment (marioba) [reply android/res/layout/list_projects.xml]: +id tells android to add a new id in the resources so you can call it from java with `R.`

**Pull request merged**